app/models/mixins/supports_feature_mixin.rb

```diff
@@ -51,18 +47,16 @@
 module SupportsFeatureMixin
   extend ActiveSupport::Concern
 
-  COMMON_FEATURES = %i[create delete destroy refresh_ems update].freeze
-
   # Whenever this mixin is included we define all features as unsupported by default.
   # This way we can query for every feature
   included do
     private_class_method :unsupported
     private_class_method :unsupported_reason_add
-    private_class_method :define_supports_feature_methods
+    class_attribute :supports_features, :instance_writer => false, :default => {}
   end
 
-  def self.reason_or_default(reason)
-    reason.present? ? reason : _("Feature not available/supported")
+  def self.reason_or_default(reason = nil)
+    reason.presence || _("Feature not available/supported")
```

Comment:
I'm not sure why this method should take a parameter at all - the caller can take care of it in all cases (and it only seems to be used here and in support_helper?

Thinking we should change this to more simply:
```suggestion
  def self.default_reason
    _("Feature not available/supported")
```

Then just change the callers from `SupportsFeatureMixin.reason_or_default(reason)` to `reason.presence || SupportsFeatureMixin.default_reason` or perhaps `reason ||= SupportsFeatureMixin.default_reason`